### ISSUE TITLE
List of packages yet to adopt artifacts

### ISSUE THREAD

**Issue by ViralBShah** [JuliaPackaging/Yggdrasil#653]
This is based on a script @pfitzseb wrote. These are packages that do not use `jll` packages in 1.3+, but depend on BinaryProvider. This does not include packages that use their own binary download mechanisms.

*Must have group*

- [x] [Blosc](https://github.com/stevengj/Blosc.jl.git)
- [x] [Cbc](https://github.com/JuliaOpt/Cbc.jl.git)
- [x] [Clp](https://github.com/JuliaOpt/Clp.jl.git) [*issue](https://github.com/JuliaPackaging/Yggdrasil/issues/509)
- [x] [Cubature](https://github.com/stevengj/Cubature.jl.git) [*issue](https://github.com/stevengj/Cubature.jl)
- [ ] [Cxx](https://github.com/JuliaInterop/Cxx.jl.git) 
- [ ] [CxxWrap](https://github.com/JuliaInterop/CxxWrap.jl.git) [*specifically libcxxwrap-julia](https://github.com/JuliaInterop/libcxxwrap-julia/issues/51)
- [x] [DecFP](https://github.com/JuliaMath/DecFP.jl.git) [*PR](https://github.com/JuliaMath/DecFP.jl/pull/106)
- [x] Elemental 
- [x] [ECOS](https://github.com/JuliaOpt/ECOS.jl.git)
- [x] [FastTransforms](https://github.com/JuliaApproximation/FastTransforms.jl.git) - Still uses BinaryProvider in Compat
- [x] [Fontconfig](https://github.com/JuliaGraphics/Fontconfig.jl.git) ([PR](https://github.com/JuliaGraphics/Fontconfig.jl/pull/31))
- [x] [GLPK](https://github.com/JuliaOpt/GLPK.jl.git)
- [x] [GSL](https://github.com/JuliaMath/GSL.jl.git) [*PR](https://github.com/JuliaMath/GSL.jl/pull/104)
- [x] [Ipopt](https://github.com/JuliaOpt/Ipopt.jl.git)
- [ ] [LibCURL](https://github.com/JuliaWeb/LibCURL.jl.git)
- [x] [LightXML](https://github.com/JuliaIO/LightXML.jl.git)  [* PR](https://github.com/JuliaIO/LightXML.jl/pull/107)
- [x] [MySQL](https://github.com/JuliaDatabases/MySQL.jl.git)  *** BP only for <1.3
- [ ] [NLopt](https://github.com/JuliaOpt/NLopt.jl.git)
- [ ] [NNlib](https://github.com/FluxML/NNlib.jl.git)
- [x] [PCRE2](https://github.com/JuliaString/PCRE2.jl.git) Stdlib. Probably should be removed from General.
- [x] [PROPACK](https://github.com/JuliaSmoothOptimizers/PROPACK.jl.git) - Still uses BinaryProvider in Compat
- [x] [SQLite](https://github.com/JuliaDatabases/SQLite.jl.git)  *** BP only for < 1.3
- [ ] [StringEncodings](https://github.com/JuliaStrings/StringEncodings.jl.git)
- [x] [SuiteSparse](https://github.com/JuliaLinearAlgebra/SuiteSparse.jl.git) Stdlib. Should probably be [removed from General](https://github.com/JuliaLinearAlgebra/SuiteSparse.jl/issues/6)
- [x] [Sundials](https://github.com/JuliaDiffEq/Sundials.jl.git) [*PR](https://github.com/SciML/Sundials.jl/pull/235)
- [x] [SCS](https://github.com/JuliaOpt/SCS.jl.git)
- [ ] [SDPA](https://github.com/JuliaOpt/SDPA.jl.git)
- [x] [SymEngine](https://github.com/symengine/SymEngine.jl.git)
- [ ] [TextAnalysis](https://github.com/JuliaText/TextAnalysis.jl.git)
- <s>[ ] [Turing](https://github.com/TuringLang/Turing.jl.git)</s> doesn't used BinaryProvider at all, unused dependency removed in https://github.com/TuringLang/Turing.jl/pull/1182
- [x] [XGBoost](https://github.com/dmlc/XGBoost.jl.git) [*PR](https://github.com/dmlc/XGBoost.jl/pull/84)

*Nice to have*

- [ ] [AMDGPUnative](https://github.com/JuliaGPU/AMDGPUnative.jl.git)
- [ ] [ArbNumerics](https://github.com/JeffreySarnoff/ArbNumerics.jl.git)
- [ ] [BayesOpt](https://github.com/jbrea/BayesOpt.jl.git)
- [x] [CALCEPH](https://github.com/JuliaAstro/CALCEPH.jl.git)
- [ ] [CDDLib](https://github.com/JuliaPolyhedra/CDDLib.jl.git)
- [ ] [CSFML](https://github.com/JuliaMultimedia/CSFML.jl.git)
- [ ] [CUDD](https://github.com/sisl/CUDD.jl.git)
- [ ] [Chemfiles](https://github.com/chemfiles/Chemfiles.jl.git)
- [ ] [Clipper](https://github.com/JuliaGeometry/Clipper.jl.git)
- [ ] [CodecXz](https://github.com/bicycle1885/CodecXz.jl.git)
- [x] [Cuba](https://github.com/giordano/Cuba.jl.git)
- [ ] [Darknet](https://github.com/ianshmean/Darknet.jl.git)
- [ ] [Deldir](https://github.com/robertdj/Deldir.jl.git)
- [ ] [Dierckx](https://github.com/kbarbary/Dierckx.jl.git)
- [ ] [ERFA](https://github.com/JuliaAstro/ERFA.jl.git)
- [ ] [EarCut](https://github.com/JuliaGeometry/EarCut.jl.git)
- [ ] [FIGlet](https://github.com/kdheepak/FIGlet.jl.git)
- [ ] [FITSIO](https://github.com/JuliaAstro/FITSIO.jl.git)
- [ ] [GithubMarkdown](https://github.com/pfitzseb/GithubMarkdown.jl.git)
- [ ] [Gmsh](https://github.com/JuliaFEM/Gmsh.jl.git)
- [ ] [GmshTools](https://github.com/shipengcheng1230/GmshTools.jl.git)
- [ ] [GoogleCodeSearch](https://github.com/tanmaykm/GoogleCodeSearch.jl.git)
- [ ] [Gumbo](https://github.com/JuliaWeb/Gumbo.jl.git)
- [ ] [H3](https://github.com/wookay/H3.jl.git)
- [ ] [HSL](https://github.com/JuliaSmoothOptimizers/HSL.jl.git)
- [ ] [Hanabi](https://github.com/JuliaReinforcementLearning/Hanabi.jl.git)
- [ ] [Hwloc](https://github.com/JuliaParallel/Hwloc.jl.git)
- [ ] [ITK](https://github.com/cj-mclaughlin/ITK.jl.git)
- [ ] [InstantiateFromURL](https://github.com/QuantEcon/InstantiateFromURL.jl.git)
- [ ] [JSONSchema](https://github.com/fredo-dedup/JSONSchema.jl.git)
- [ ] [KaTeX](https://github.com/piever/KaTeX.jl.git)
- [ ] [LSL](https://github.com/samuelpowell/LSL.jl.git)
- [ ] [LazIO](https://github.com/evetion/LazIO.jl.git)
- [ ] [LibPQ](https://github.com/invenia/LibPQ.jl.git)
- [ ] [LibSpatialIndex](https://github.com/JuliaGeo/LibSpatialIndex.jl.git)
- [ ] [LibSymspg](https://github.com/unkcpz/LibSymspg.jl.git)
- [ ] [LibYAML](https://github.com/eschnett/LibYAML.jl.git)
- [ ] [Libtask](https://github.com/TuringLang/Libtask.jl.git)
- [ ] [Libxc](https://github.com/unkcpz/Libxc.jl.git)
- [ ] [MFrontInterface](https://github.com/JuliaFEM/MFrontInterface.jl.git)
- [ ] [MakieGallery](https://github.com/JuliaPlots/MakieGallery.jl.git)
- [ ] [Mongoc](https://github.com/felipenoris/Mongoc.jl.git)
- [ ] [Nemo](https://github.com/Nemocas/Nemo.jl.git)
- [ ] [Nettle](https://github.com/staticfloat/Nettle.jl.git) [*issue](https://github.com/staticfloat/Nettle.jl/issues/103)
- [ ] [Nuklear](https://github.com/Gnimuc/Nuklear.jl.git)
- [ ] [OSQP](https://github.com/oxfordcontrol/OSQP.jl.git)
- [ ] [Polymake](https://github.com/oscar-system/Polymake.jl.git)
- [ ] [RNGTest](https://github.com/andreasnoack/RNGTest.jl.git)
- [ ] [Remark](https://github.com/piever/Remark.jl.git)
- [ ] [RingBuffers](https://github.com/JuliaAudio/RingBuffers.jl.git)
- [ ] [SDPAFamily](https://github.com/ericphanson/SDPAFamily.jl.git)
- [ ] [SPICE](https://github.com/JuliaAstro/SPICE.jl.git)
- [ ] [Sass](https://github.com/piever/Sass.jl.git)
- [ ] [Shoco](https://github.com/ararslan/Shoco.jl.git)
- [ ] [Singular](https://github.com/oscar-system/Singular.jl.git)
- [ ] [Snappy](https://github.com/bicycle1885/Snappy.jl.git)
- [ ] [SodiumSeal](https://github.com/christopher-dG/SodiumSeal.jl.git)
- [ ] [SoftPosit](https://github.com/milankl/SoftPosit.jl.git)
- [ ] [Spot](https://github.com/sisl/Spot.jl.git)
- [ ] [StringAnalysis](https://github.com/zgornel/StringAnalysis.jl.git)
- [ ] [TetGen](https://github.com/JuliaGeometry/TetGen.jl.git)
- [ ] [Thrift](https://github.com/tanmaykm/Thrift.jl.git)
- [ ] [Triangulate](https://github.com/JuliaGeometry/Triangulate.jl.git)
- [ ] [ViZDoom](https://github.com/JuliaReinforcementLearning/ViZDoom.jl.git)
- [ ] [WCS](https://github.com/JuliaAstro/WCS.jl.git)
- [ ] [WebAssembly](https://github.com/MikeInnes/WebAssembly.jl.git)
- [ ] [XXhash](https://github.com/hros/XXhash.jl.git)
- [ ] [Xsum](https://github.com/stevengj/Xsum.jl.git)
- [ ] [YAJL](https://github.com/christopher-dG/YAJL.jl.git)
- [ ] [YOLO](https://github.com/ianshmean/YOLO.jl.git)

**Comment by ViralBShah** [comment]
This list is not perfect, and includes

1. Stdlibs that will become JLLs
2. Use JLLs on master but need a release
3. Still have BinaryProvider in project file but don't actually use it
4. Actually still depend on BinaryProvider on master
5. Depends on BinaryProvider to support Julia versions older than 1.2
6. Work in progress packages that are being prepared for artifacts

Also, some packages are more widely used than others, so perhaps it would be good to split that list into "must have" and "nice to have".

**Comment by j-fu** [comment]
Hi, Triangulate.jl is ready for it, but how to organize things to keep supporting Julia 1.0 ? Would also care about TetGen if time permits.

**Comment by ViralBShah** [comment]
Many packages have moved to 1.3+ so that they only support artifacts. I think in the spirit of moving forward, that is the right thing to do.

**Comment by ViralBShah** [comment]
@mlubin

**Comment by odow** [comment]
The COIN solvers Cbc, Clp, and Ipopt have exceedingly complicated build systems (https://github.com/JuliaPackaging/Yggdrasil/issues/509). I have no estimate for how much work is required to make it work. @juan-pablo-vielma managed to make the current binary builders work only after hours and hours of trial and error, and packaging everything up into one giant binary.

Is this a "would be nice to use jll's" or will the current binary builder implementations stop working on a future 1.x release?!?

GLPK is probably not too hard. I'll take a look.

**Comment by giordano** [comment]
Let's continue the discussion about Clp in #509, no need to clutter this issue

**Comment by mlubin** [comment]
@ViralBShah The answer appears to be no, there isn't currently an effort. Is BinaryBuilder being deprecated?

**Comment by giordano** [comment]
> s BinaryBuilder being deprecated?

BinaryProvider, not BinaryBuilder: https://julialang.org/blog/2019/11/artifacts/

**Comment by ViralBShah** [comment]
@StefanKarpinski Bringing this list to your notice.

**Comment by SimonDanisch** [comment]
Could we try to semi automatically move some of these to Yggdrasil? E.g. look up the binarybuilder URL from the `deps.jl`, download the the `build_tarballs.jl` and add it to the correct folder?

**Comment by ViralBShah** [comment]
> Could we try to semi automatically move some of these to Yggdrasil? E.g. look up the binarybuilder URL from the `deps.jl`, download the the `build_tarballs.jl` and add it to the correct folder?

I've done that process manually in many cases. Given that there's only probably 2 dozen more such libraries, perhaps not worth implementing tooling.

**Comment by jaakkor2** [comment]
Since MathOptInterface depends on JSONSchema, I would re-classify JSONSchema from "nice to have" to "must have".

**Comment by odow** [comment]
JSONSchema doesn't actually use BinaryProvider to for binaries. It is just used to download files in a test. @fredo-dedup isn't very active on Github, but I have commit access, and was in the process of tidying up the tests: https://github.com/fredo-dedup/JSONSchema.jl/pull/13. I should remove the dependency on BinaryProvider while I'm at it.

**Comment by ViralBShah** [comment]
Enough of the commonly used packages are done, that I feel we can close this issue.

**Comment by ViralBShah** [comment]
Updated that in the list above.

**Comment by aviks** [comment]
[NLOpt](https://github.com/JuliaOpt/NLopt.jl/pull/145/files) is also done

**Comment by ViralBShah** [comment]
Filed https://github.com/JuliaStrings/StringEncodings.jl/issues/36 for StringEncodings.jl. Should be straighforward to update the package.

@aviks I also marked TextAnalysis done.

**Comment by ViralBShah** [comment]
I believe TextAnalysis and StringAnalysis both need libstemmer.

**Comment by aviks** [comment]
[Mongoc](https://github.com/felipenoris/Mongoc.jl/pull/67) is done.

**Comment by ViralBShah** [comment]
The big missing one now is GR, which has a dependency on QT.

**Comment by ViralBShah** [comment]
GR is done now.
